### PR TITLE
The backup cleaner can't clean the CR at the `schedule` phase before task start

### LINKED ISSUE #5968: The backup cleaner can't clean the CR at the schedule phase before task start
## Bug Report

**What version of Kubernetes are you using?**

default

**What version of TiDB Operator are you using?**

master

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

not relevant

**What's the status of the TiDB cluster pods?**

not relevant

**What did you do?**

1. Start a backup job
2. The backup job stuck at the `schedule` phase for some reason
3. Try to delete the CR

**What did you expect to see?**

The CR should be cleaned as expect

**What did you see instead?**

The cleaner will skip the cleaning and nothing happened.

### PR DESCRIPTION
**Pull request by RidRisR**
### What problem does this PR solve?

Close #5968 

### What is changed and how does it work?

Set the status of the not started CR to stop phase.
The task haven't been started so there will ok to delete it immediately.

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: tests/e2e/br/br.go [hunk 1]
```diff
@@ -667,6 +667,72 @@ var _ = ginkgo.Describe("Backup and Restore", func() {
 			framework.ExpectEqual(cleaned, true, "storage should be cleaned")
 		})
 
+		ginkgo.It("start,truncate,stop log backup using old interface", func() {
```
Comment (csuzhangxc) [inline tests/e2e/br/br.go:670]: It seems this case name is not accurate. We are stoping a scheduled task, right?

Comment (RidRisR) [reply tests/e2e/br/br.go]: done

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**